#!/usr/bin/env python
PACKAGE = "tpo_vision"
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name       Type      Reconfiguration level    Description           Default   Min   Max
gen.add("hue_min",    int_t,    0, "Hue Min for the first threshold", 124,  0, 255)
gen.add("hue_max",    int_t,    0, "Hue Max for the first threshold", 255,  0, 255)
gen.add("sat_min",    int_t,    0, "Sat Min for the first threshold", 60,  0, 255)
gen.add("sat_max",    int_t,    0, "Sat Max for the first threshold", 164,  0, 255)
gen.add("val_min",    int_t,    0, "Value Min for the first threshold", 189,  0, 255)
gen.add("val_max",    int_t,    0, "Value Min for the first threshold", 255,  0, 255)

gen.add("thresh_gray_min",    int_t,    0, "Min for the second threshold", 100,  0, 255)
gen.add("thresh_gray_max",    int_t,    0, "Max for the second threshold", 255,  0, 255)
gen.add("blob_param_color",    int_t,    0, "Color filter for blob detection", 255,  0, 255)

gen.add("median_blur_ksize",    int_t,    0, "aperture linear size for openxc.medianblur()", 0,  0, 600)
gen.add("contrast",    double_t,    0, "contrast", 1,  0, 20)
gen.add("brightness",    double_t,    0, "brightness", 0,  -255, 255)


exit(gen.generate(PACKAGE, "LaserSpotDetectionNode", "LaserSpotDetectionParams"))
